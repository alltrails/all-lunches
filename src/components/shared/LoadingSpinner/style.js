import styled, { keyframes } from 'styled-components';

const ringspin = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const RingSpinner = styled.div`
  display: inline-block;
  position: relative;
  vertical-align: sub;

  /* Control the overall diameter of the circle here */
  width: 2.2rem;
  height: 2.2rem;

  & div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 100%;
    height: 100%;
    border-radius: 50%;

    /* Control the width of the circular line here */
    border: 3px solid;

    /* Make different border sides transparent for different patterns */
    ${({ color }) => `
      border-color: ${color} ${color} transparent transparent;
    `}

    /* Changing cubic-bezier
     * See http://cubic-bezier.com to make easy variants
     */
    animation: ${ringspin} 1.2s cubic-bezier(.57,.05,.58,.94) infinite;
  }

  & div:nth-child(1) {
    animation-delay: -0.25s;
  }
  & div:nth-child(2) {
    animation-delay: -0.17s;
  }
  & div:nth-child(3) {
    animation-delay: -0.1s;
  }
`;
