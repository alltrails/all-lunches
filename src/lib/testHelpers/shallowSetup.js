import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
