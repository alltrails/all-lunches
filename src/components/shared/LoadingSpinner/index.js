import React from 'react';
import PropTypes from 'prop-types';

import colors from 'styles/colors';

import { RingSpinner } from './style';

const LoadingSpinner = ({ color }) => (
  <RingSpinner color={color}>
    <div />
    <div />
    <div />
    <div />
  </RingSpinner>
);

LoadingSpinner.propTypes = {
  color: PropTypes.string,
};

LoadingSpinner.defaultProps = {
  color: colors.blue,
};

export default LoadingSpinner;
