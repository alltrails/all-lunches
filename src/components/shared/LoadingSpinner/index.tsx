import React from 'react';
import colors from 'styles/colors';

import { RingSpinner } from './style';

type LoadingSpinnerProps = {
  color?: string;
};

const LoadingSpinner = ({ color = colors.green }: LoadingSpinnerProps) => (
  <RingSpinner color={color}>
    <div />
    <div />
    <div />
    <div />
  </RingSpinner>
);

export default LoadingSpinner;
