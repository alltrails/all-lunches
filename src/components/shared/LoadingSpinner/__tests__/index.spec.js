/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import LoadingSpinner from '../index';

describe('LoadingSpinner', () => {
  const defaultProps = {};
  const setup = shallowSetup(LoadingSpinner, defaultProps);

  it('renders the expected view with no color option', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view with a color option', () => {
    const { wrapper } = setup({ color: '#fff' });
    expect(wrapper).toMatchSnapshot();
  });
});
