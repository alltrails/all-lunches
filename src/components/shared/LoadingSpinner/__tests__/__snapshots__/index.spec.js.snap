// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders the expected view with a color option 1`] = `
<style__RingSpinner
  color="#fff"
>
  <div />
  <div />
  <div />
  <div />
</style__RingSpinner>
`;

exports[`LoadingSpinner renders the expected view with no color option 1`] = `
<style__RingSpinner
  color="#428813"
>
  <div />
  <div />
  <div />
  <div />
</style__RingSpinner>
`;
